### PR TITLE
GitHub Action to validate a Helm chart using Helm template

### PR DESCRIPTION
**Pull request by 00xkhaled**
## What

GitHub Action to validate a Helm chart using Helm template

## Why
DEVOPS-892

## References
DEVOPS-892

### CODE REVIEW

File: helm-chart-validation/action.yml [hunk 1]
```diff
@@ -34,9 +37,10 @@ runs:
       uses: greenbone/actions/set-github-user@v3
 
     - name: Clone Repository
+      if: inputs.repository-url && inputs.repository-branch
       shell: bash
       run: |
-        git clone ${{ inputs.repository-url }}
+        git clone --branch ${{ inputs.repository-branch }} ${{ inputs.repository-url }}
```
Comment (y0urself) [inline helm-chart-validation/action.yml:43]: Use the [`actions/checkout` action](https://github.com/actions/checkout) for this. ;)

File: helm-chart-validation/action.yml [hunk 2]
```diff
@@ -37,10 +37,10 @@ runs:
       uses: greenbone/actions/set-github-user@v3
 
     - name: Clone Additional Repository
-      if: inputs.repository-url && inputs.repository-branch
+      if: inputs.repository-name && inputs.repository-branch
       uses: actions/checkout@v2
```
Comment (y0urself) [inline helm-chart-validation/action.yml:41]: You need to clone both the "working" repository and the `greenbone/product-helm-chart` (into separate directories) by using that action twice!

```suggestion
      uses: actions/checkout@v4
```

**Review by bjoernricks — COMMENTED** [review]
There is no need to put an additional LICENSE file into this action directory

File: helm-chart-validation/action.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    required: false
+  additional-chart-values-path:
+    description: "Path to the additional values.yaml file within the cloned repository (optional)"
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Git credentials 
+      uses: greenbone/actions/set-github-user@v3
```
Comment (y0urself) [inline helm-chart-validation/action.yml:37]:
```suggestion
    - name: Checkout repo
      uses: actions/checkout@v4
```

**Review by pascalholthaus — COMMENTED** [review]
Looks fine for me.

**Review by pascalholthaus — APPROVED** [review]

**Pull request merged**